clustercontroller-utils/src/main/java/com/yahoo/vespa/clustercontroller/utils/staterestapi/server/JsonReader.java

```diff
@@ -98,7 +109,33 @@ public SetRequestData getStateRequestData(HttpRequest request) throws Exception
             }
             stateMap.put(type, new UnitStateImpl(code, reason));
         }
-        return new SetRequestData(stateMap, condition, responseWait);
+
+        final Optional<Duration> timeout = parseTimeout(request.getOption("timeout", null));
+
+        return new SetRequestData(stateMap, condition, responseWait, timeout);
+    }
+
+    public static Optional<Duration> parseTimeout(String timeoutOption) throws InvalidContentException {
```

Comment:
Since this is a public method, consider moving out to a separate utility class. Request parameter parsing implementation smells like it should live in a different class than JSON reading.

Reply:
I'll move timeout parsing to RestApiHandler.